index.bs

```diff
@@ -1046,101 +1012,480 @@ BigIntValue = {
+
+      1. Return [=success=] with data [=Object.fromEntries=](|iterable|).
+
+    <dt>|type| is the string "<code>regexp</code>"
+    <dd>
+      1. Let |pattern| be the value of the <code>pattern</code> field of |local protocol value|.
+
+      1. Let |flags| be the value of the <code>flags</code> field of |local protocol value| or
+         undefined if no such a field.
+
+      1. Let |regex_result| be [=Regexp=](|pattern|, |flags|). If this throws exception, return [=error=]
+         with [=error code=] [=invalid argument=].
+
+      1. Return [=success=] with data |regex_result|.
+
+    <dt>|type| is the string "<code>set</code>"
+    <dd>
+      1. Let |deserialized value list| be a result of [=trying=] to [=deserialize value list=]
+         with |value|.
+
+      1. Let |iterable| be [=CreateArrayFromList=](|deserialized key-value list|)
+
+      1. Return [=success=] with data [=Set object=](|iterable|).
+
+    <dt>otherwise
+    <dd>Return [=error=] with [=error code=] [=invalid argument=].
+  </dl>
+
+</div>
+
+### Remote Value ### {#data-types-protocolValue-RemoteValue}
+
+Values accessible from the ECMAScript runtime are represented by a mirror
+object, specified as <code>RemoteValue</code>. The value's type is specified in
+the <code>type</code> property. In the case of JSON-representable primitive
+values, this contains the value in the <code>value</code> property; in the case
+of non-JSON-representable primitives, the <code>value</code> property contains a
+string representation of the value. For non-primitive objects, the
+<code>objectId</code> property contains a string id that provides a unique
+handle to the object, valid for its lifetime inside the engine. For some
+non-primitive types, the <code>value</code> property contains a representation
+of the data in the ECMAScript object; for container types this can contain
+further <code>RemoteValue</code> instances. The <code>value</code> property can
+be null if there is a duplicate object i.e. the object has already been
+serialized in the current <code>RemoteValue</code>, perhaps as part of a
+cycle, or otherwise when the maximum serialization depth is reached.
+
+[=Nodes=] are also represented by <code>RemoteValue</code> instances. These have
+a partial serialization of the node in the value property.
+
+Issue: reconsider mirror objects' lifecycle.
+
+Note: mirror objects do not keep the original object alive in the runtime. If an
+object is discarded in the runtime, subsequent attempts to access it via the
+protocol will result in an error.
+
+<div algorithm>
+To get the <dfn>object id for an object</dfn> given a |session| and |object|:
+
+1. If |session|'s [=object id map=] does not contain |object|, run the
+   following steps:
+
+  1. Let |object id| be a new, unique, string identifier for |object|. If
+     |object| is an [=/element=] this must be the [=web element reference=] for
+     |object|; if it's a {{WindowProxy}} object, this must be the [=window
```

Comment:
What about a `shadow root identifier`? It seems to be missing here.

I'm not sure if we still need a `frame identifier`. @jgraham?

Reply:
@whimboo I'm not sure how to handle the shadow roots. Could you please create a dedicated issue to keep the question out of scope of the current PR?

Reply:
I would have done but missed to do that because this issue was marked as resolved. Filed https://github.com/w3c/webdriver-bidi/issues/156 now.